From fe5679ef12dae27519b4e675a580ae2ae5cda595 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Cl=C3=A9ment=20P=C3=A9ron?= <peron.clem@gmail.com>
Date: Tue, 21 May 2019 18:11:02 +0200
Subject: [PATCH] arm64: dts: allwinner: Add mali GPU supply for H6 boards
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Enable and add supply to the Mali GPU node on all the
H6 boards.

Regarding the datasheet the maximum time for supply to reach
its voltage is 32ms.

Signed-off-by: Clément Péron <peron.clem@gmail.com>
---
 arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts | 6 ++++++
 arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi-3.dts  | 6 ++++++
 arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi.dtsi   | 6 ++++++
 arch/arm64/boot/dts/allwinner/sun50i-h6-pine-h64.dts    | 6 ++++++
 4 files changed, 24 insertions(+)

diff -Naur linux-5.2.3-old/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts linux-5.2.3-new/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts
--- linux-5.2.3-old/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts	2019-07-26 09:11:12.000000000 +0200
+++ linux-5.2.3-new/arch/arm64/boot/dts/allwinner/sun50i-h6-beelink-gs1.dts	2019-07-26 16:19:57.946666666 +0200
@@ -70,6 +70,11 @@
 	status = "okay";
 };
 
+&gpu {
+	mali-supply = <&reg_dcdcc>;
+	status = "okay";
+};
+
 &hdmi {
 	status = "okay";
 };
@@ -206,6 +211,7 @@
 			};
 
 			reg_dcdcc: dcdcc {
+				regulator-enable-ramp-delay = <32000>;
 				regulator-min-microvolt = <810000>;
 				regulator-max-microvolt = <1080000>;
 				regulator-name = "vdd-gpu";

diff -Naur linux-5.2.3-old/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi-3.dts linux-5.2.3-new/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi-3.dts
--- linux-5.2.3-old/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi-3.dts	2019-07-26 16:07:22.600000002 +0200
+++ linux-5.2.3-new/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi-3.dts	2019-07-26 16:19:57.946666666 +0200
@@ -153,6 +153,11 @@
 	status = "okay";
 };
 
+&gpu {
+	mali-supply = <&reg_dcdcc>;
+	status = "okay";
+};
+
 &hdmi {
 	status = "okay";
 };
@@ -309,6 +314,7 @@
 			};
 
 			reg_dcdcc: dcdcc {
+				regulator-enable-ramp-delay = <32000>;
 				regulator-min-microvolt = <810000>;
 				regulator-max-microvolt = <1080000>;
 				regulator-name = "vdd-gpu";
diff -Naur linux-5.2.3-old/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi.dtsi linux-5.2.3-new/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi.dtsi
--- linux-5.2.3-old/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi.dtsi	2019-07-26 09:11:12.000000000 +0200
+++ linux-5.2.3-new/arch/arm64/boot/dts/allwinner/sun50i-h6-orangepi.dtsi	2019-07-26 16:19:57.946666666 +0200
@@ -55,6 +55,11 @@
 	status = "okay";
 };
 
+&gpu {
+	mali-supply = <&reg_dcdcc>;
+	status = "okay";
+};
+
 &mmc0 {
 	vmmc-supply = <&reg_cldo1>;
 	cd-gpios = <&pio 5 6 GPIO_ACTIVE_LOW>;
@@ -163,6 +168,7 @@
 			};
 
 			reg_dcdcc: dcdcc {
+				regulator-enable-ramp-delay = <32000>;
 				regulator-min-microvolt = <810000>;
 				regulator-max-microvolt = <1080000>;
 				regulator-name = "vdd-gpu";
diff -Naur linux-5.2.3-old/arch/arm64/boot/dts/allwinner/sun50i-h6-pine-h64.dts linux-5.2.3-new/arch/arm64/boot/dts/allwinner/sun50i-h6-pine-h64.dts
--- linux-5.2.3-old/arch/arm64/boot/dts/allwinner/sun50i-h6-pine-h64.dts	2019-07-26 16:07:22.286666669 +0200
+++ linux-5.2.3-new/arch/arm64/boot/dts/allwinner/sun50i-h6-pine-h64.dts	2019-07-26 16:19:57.946666666 +0200
@@ -89,6 +89,11 @@
 	mali-supply = <&reg_dcdcc>;
 };
 
+&gpu {
+	mali-supply = <&reg_dcdcc>;
+	status = "okay";
+};
+
 &hdmi {
 	status = "okay";
 };
@@ -219,6 +224,7 @@
 			};
 
 			reg_dcdcc: dcdcc {
+				regulator-enable-ramp-delay = <32000>;
 				regulator-min-microvolt = <810000>;
 				regulator-max-microvolt = <1080000>;
 				regulator-name = "vdd-gpu";
